### PR TITLE
NC-607

### PR DESCRIPTION
**Pull request by Joeper214**
ref: https://skyarch.backlog.jp/view/NC-607

### CODE REVIEW

**Comment by Joeper214** [comment]
4 more errors on test. 😖

**Comment by root-beer** [comment]
Let's investigate. 👮

**Comment by Joeper214** [comment]
ありがとうございます。

File: spec/controllers/users_admin_controller_spec.rb [hunk 1]
```diff
@@ -15,11 +15,11 @@
   run_zabbix_server
 
   let(:klass){User}
-  let(:zabbix_servers){['', create(:zabbix_server)]}
+  let(:zabbix_servers){[create(:zabbix_server)]}
   let(:user){create(:user)}
   let(:admin_status){false}
   let(:master_status){false}
-  let(:user_hash) { attributes_for(:user, admin: admin_status, master: master_status, zabbix_servers: zabbix_servers) }
+  let(:user_hash) { attributes_for(:user, admin: admin_status, master: master_status, zabbix_servers: zabbix_servers.unshift('')) }
```
Comment (Joeper214) [inline spec/controllers/users_admin_controller_spec.rb]: 👍

**Comment by Joeper214** [comment]
ありがとうございます。@root-beer.

Maybe you can Review and Merge this PR. 

I'll add more feature in the next PR.

🙇

**Comment by root-beer** [comment]
I'll review now. 😃

File: app/assets/javascripts/zabbix_servers.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          return this.picked.delete_zabbix_server_path ? true : false;
+      },
+      can_edit: function() {
+        if (this.picked.edit_zabbix_server_url)
+          return this.picked.edit_zabbix_server_url ? true : false;
+      },
+
+      delete_entry: function()  {
+        var self = this;
+        modal.Confirm(t('users.user'), t('users.msg.delete_user', self.email), 'danger').done(function () {
```
Comment (root-beer) [inline app/assets/javascripts/zabbix_servers.js]: It seems wrong. (user?)

Comment (Joeper214) [inline app/assets/javascripts/zabbix_servers.js]: すみません。i'll change it now.

File: app/models/user.rb [hunk 3]
```diff
@@ -28,6 +28,9 @@ class User < ActiveRecord::Base
   extend Concerns::Cryptize
   cryptize :mfa_secret_key
 
+  attr_accessor :zabbix_servers
+  serialize :zabbix_servers, JSON
```
Comment (root-beer) [inline app/models/user.rb]: There seems to be no `zabbix_servers` column on the `users` table.

Comment (Joeper214) [inline app/models/user.rb]: I added it as a dummy attribute. Can I add it on the next branch? Because this branch seems to be huge already.

Comment (root-beer) [inline app/models/user.rb]: I see. It's okay. 👍

Comment (Joeper214) [inline app/models/user.rb]: One of the reason is that `zabbix_servers` has no connection to users yet. So I'm planning to add it on the `users` table but only `zabbix_server_ids` is requried.

Comment (Joeper214) [inline app/models/user.rb]: Currenty, the concern here is that when the user is created and selects many or one `zabbix_servers` it needs to be inserted on the `user` table. And if the user deletes a `user` it should automatically be removed from the selected `zabbix_servers`. So, I'm planning to the implementation on the next branch. 🙇

Comment (Joeper214) [inline app/models/user.rb]: So, maybe I need to create a new ERD diagram for this right? @root-beer

Comment (root-beer) [inline app/models/user.rb]: I understand it.
I'll leave it to you. 🙇

**Comment by root-beer** [comment]
LGTM! Thanks. @Joeper214 🙇

**Comment by Joeper214** [comment]
🙇

**Pull request merged**